### PR TITLE
Use Docker CLI socket from home

### PR DESCRIPTION
**Pull request by p1-0tr**
**What I did**

We should not rely on having a global path for the Docker CLI socket. On
macOS this forces Docker Desktop to access directories which require
raised privileges. Whereas on Linux we do not create sockets in that
location at all, currently. So look for the Docker CLI socket in the
User's home directory.

**Related issue**

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: cli/metrics/conn_other.go
Comment (thaJeztah) [inline cli/metrics/conn_other.go:5]: I think the validation currently doesn't allow for the years to differ between files, so all should have the same years

File: cli/metrics/conn_darwin.go
Comment (p1-0tr) [reply cli/metrics/conn_darwin.go]: Good point :) thanks

File: cli/metrics/conn_other.go
Comment (p1-0tr) [reply cli/metrics/conn_other.go]: Yeah :( I even did a quick google for some more decent copyright header checker, but couldn't find one which does the right thing

Comment (thaJeztah) [reply cli/metrics/conn_other.go]: In https://github.com/docker/cli-docs-tool/, we've been trying Google addlicense, which seemed to work (a bit more "loosely" in checking); see https://github.com/docker/cli-docs-tool/pull/4

File: .gitignore
Comment (mat007) [inline .gitignore:5]: Shouldn’t all that be in your global gitignore on your machine?

Comment (p1-0tr) [reply .gitignore:5]: Sure, that will work as well. Putting those in project-specific configs is force of habit for me.

File: cli/metrics/conn_windows.go
Comment (mat007) [inline cli/metrics/conn_windows.go:35]: Did you mean
```suggestion
	overrideSocket() // no-op, unless built for e2e testing
```
or something like that?

**Review by glours — APPROVED** [review]
Sounds good to me, I'm just waiting for an approval from someone of the Pos team to merge it

File: cli/metrics/conn_windows.go
Comment (p1-0tr) [reply cli/metrics/conn_windows.go:35]: `nop` is the mnemonic for no-op in most assemblers (and tends to be used as a neologism for "do nothing"), but yeah that would be clearer.

Comment (p1-0tr) [reply cli/metrics/conn_windows.go]: done :)

File: cli/metrics/conn_darwin.go
Comment (djs55) [inline cli/metrics/conn_darwin.go:36]: Maybe we should use a shorter Linux-style path in `$HOME/.docker/desktop/docker-cli.sock` even on Darwin because we've had some problems where the maximum path in the sockaddr is 104 characters (even shorter than Linux's 108 characters). We work around this in some of our binaries by calling `os.Chdir("Library/Containers/com.docker.docker")` and then using a relative path... but I suspect this trick doesn't work for the CLI because changing the current directory probably breaks something.

WDYT?

Comment (p1-0tr) [reply cli/metrics/conn_darwin.go:36]: Hmm, this is a tough one.

The best thing I can think of from the top of my head is:

    pushd ...
    connect with relative path
    popd
    
but not sure that would be safe to do.

Comment (djs55) [reply cli/metrics/conn_darwin.go:36]: Reflecting on it a bit, I'm not as concerned as I was because the socket is best-effort (and it will work probably 99% of the time). So I'm happy with it as-is.

**Review by djs55 — APPROVED** [review]

**Pull request merged**
